src/llm_utils.py

```diff
@@ -5,178 +5,136 @@
-    )   
+        Important Rules:
+        1. Never include code snippets in COMMENT
+        2. Always verify line numbers exist in the diff
+        3. Keep suggestions focused on changed lines
+        4. Use exactly one '---' between comments
 
-    #print(response.text)
-    
-    # Parse and format the response
-    review_comments = parse_llm_response(response.text)
-    return review_comments
+        Diff content:
+        {diff_content}
+        """
 
+        client = genai.Client(api_key=API_KEY)
+
+        #Get analysis from Gemini model
+        response = client.models.generate_content(
+            model="gemini-2.0-flash", contents=prompt
+        )   
+
+        #print(response.text)
+        
+        # Parse and format the response
+        review_comments = parse_llm_response(response.text)
+        print(f"Parsed {len(review_comments)} valid comments")
+        return review_comments
+    except Exception as e:
+        print(f"LLM analysis failed: {str(e)}")
+        return []
 
 
-"""
-This is the final code given by deepseek to parse the response from the LLM model. It includes errors
-"""
 
 def parse_llm_response(response: str) -> List[Dict]:
+    """Parse LLM response with strict validation"""
     review_comments = []
     
-    # Split response into valid comment blocks
-    comment_blocks = response.split('---')
-    
-    for block in comment_blocks:
-        block = block.strip()
-        if not block:
+    for block in response.split('---'):
+        lines = [line.strip() for line in block.split('\n') if line.strip()]
+        if not lines:
             continue
-            
-        comment_data = {
+
+        comment = {
             'path': None,
-            'line': None,
-            'body': ''
+            'start_line': None,
+            'end_line': None,
+            'body': []
         }
+        valid = True
         
-        # Validate required fields
-        has_file = False
-        has_line = False
-        has_comment = False
-        
-        for line in block.split('\n'):
-
-            line = line.strip()
-
+        # Parse block
+        for line in lines:
             if line.startswith('FILE:'):
-                comment_data['path'] = line.split('FILE:', 1)[-1].strip()
-                has_file = True
-
-            elif line.startswith('LINE:'):
+                comment['path'] = line.split('FILE:', 1)[-1].strip()
+            elif line.startswith('LINES:'):
                 try:
-                    comment_data['line'] = int(line.split('LINE:', 1)[-1].strip())
-                    has_line = True
+                    parts = line.split('LINES:', 1)[-1].strip().split('-')
+                    start = int(parts[0])
+                    end = int(parts[1])
+                    comment['start_line'], comment['end_line'] = sorted([start, end])
                 except (ValueError, IndexError):
-                    continue
-
+                    valid = False
```

Comment:
Parsing the line numbers
```suggestion
|
```
N/A